### DISCUSSION TITLE
Training loop for GNN nodes regression task

### DISCUSSION THREAD

**Discussion by zahattar** [pyg-team/pytorch_geometric#7903] [Q&A]
I am currently working with the PYG library on a graph neural network project focused on node regression. In order to train this network, I generate my own collection of graphs using a “for loop.” Within this loop, I create specific data objects including node features, edge indices, and target values. These data objects are then compiled into a list, which becomes my training set. To manage the data efficiently, I utilize a DataLoader.

The core architecture of my GNN model is a Graph Convolutional Network (GCN). For the training process, I am struggling with the question of how to structure the “for loop.” Specifically, I am uncertain about whether I should use batches and how best to feed the graphs into the model(should I feed the model one graph at a time? And how to code this). This aspect of the training loop is posing a challenge, and I am seeking guidance on the most effective approach.

**Comment by rusty1s** [comment]
Since you are already using `DataLoader`, you basically get mini-batching for free from PyG. You can simply specify a `batch_size>1`, and PyG will take care of creating mini-batches. Depending on the task, there is not much change needed to make your GNN work in mini-batching scenarios over single graph processing. Overall, mini-batching is preferred to speed up convergence and efficiency in general.
